rpc/rpcserver/server.go

```diff
@@ -1153,6 +1161,123 @@ func StartWalletLoaderService(server *grpc.Server, loader *loader.Loader,
 	pb.RegisterWalletLoaderServiceServer(server, service)
 }
 
+// StartTicketBuyerService creates an implementation of the TicketBuyerService
+// and registers it with the gRPC server.
+func StartTicketBuyerService(server *grpc.Server, loader *loader.Loader, tbCfg *ticketbuyer.Config) {
+	service := &ticketbuyerServer{loader: loader, ticketbuyerCfg: tbCfg}
+	pb.RegisterTicketBuyerServiceServer(server, service)
+}
+
+// StartAutoBuyer starts the automatic ticket buyer.
+func (t *ticketbuyerServer) StartAutoBuyer(ctx context.Context, req *pb.StartAutoBuyerRequest) (
+	*pb.StartAutoBuyerResponse, error) {
+
+	defer zero.Bytes(req.Passphrase)
+
+	wallet, ok := t.loader.LoadedWallet()
+	if !ok {
+		return nil, grpc.Errorf(codes.FailedPrecondition, "Wallet has not been loaded")
+	}
+	err := wallet.Unlock(req.Passphrase, nil)
+	if err != nil {
+		return nil, translateError(err)
+	}
+
+	accountName, err := wallet.AccountName(req.Account)
+	if err != nil {
+		return nil, translateError(err)
+	}
+
+	balanceToMaintain := dcrutil.Amount(req.BalanceToMaintain)
+	if balanceToMaintain < 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Negative balance to maintain given")
+	}
+
+	maxFeePerKB := dcrutil.Amount(req.MaxFeePerKb)
+
+	if maxFeePerKB < 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Negative max fee per KB given")
+	}
+
+	if req.MaxPriceAbsolute < 0 && req.MaxPriceRelative < 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Negative max ticket price given")
+	}
+	params := wallet.ChainParams()
+
+	var votingAddress string
+	if req.VotingAddress != "" {
+		votingAddr, err := decodeAddress(req.VotingAddress, params)
+		if err != nil {
+			return nil, err
+		}
+		votingAddress = votingAddr.String()
+	}
+
+	var poolAddress string
+	if req.PoolAddress != "" {
+		poolAddr, err := decodeAddress(req.PoolAddress, params)
+		if err != nil {
+			return nil, err
+		}
+		poolAddress = poolAddr.String()
+	}
+
+	poolFees := req.PoolFees
+	if poolFees != 0 || poolAddress != "" {
+		err = txrules.IsValidPoolFeeRate(poolFees)
+		if err != nil {
+			return nil, grpc.Errorf(codes.InvalidArgument,
+				"Pool fees amount invalid: %v", err)
```

Comment:
this could potentially be a confusing error, if the pool address was not the empty string.  According to the documentation, 0 is a valid value for the pool fees, but it's only in combination with a non-"" pool address where this is invalid.